js/lib/Transaction.js

```diff
@@ -140,9 +126,17 @@ class Transaction {
         return '0x' + this.hash().toString('hex')
     }
 
-    typeVals() {
+    tid() {
+        return Buffer.from(web3Utils.soliditySha3.apply(null, this.typeVals(true)).substring(2), 'hex');
+    }
+
+    tidHex() {
+        return '0x' + this.tid().toString('hex')
+    }
+
+    typeVals(includeSignatures) {
         let res = [];
-        this.inputs.forEach(input => res = res.concat(input.typeVals()));
+        this.inputs.forEach(input => res = res.concat(input.typeVals(includeSignatures)));
```

Comment:
perhaps just mapping over the `inputs` and assigning the returning array to `res` would make it more declarative, like:

```js
let res = this.inputs.map(input => input.typeVals())
```

Reply:
thanks for the trick